lib/apis/moray.js

```diff
@@ -1021,4 +1149,36 @@ Moray.prototype.delVmRoleTags = function (uuid, cb) {
     });
 };
 
+/**
+ * Reindexes all objects in the bucket if it has been updated.
+ *
+ * @param moray {MorayClient}
+ * @param bucketName {Name of the bucket to reindex}
+ * @param callback {Function} `function (err)`
+ */
+Moray.prototype._reindexBucket = function _reindexBucket(bucketName, callback) {
+    assert.string(bucketName, 'bucketName');
+    assert.func(callback, 'callback');
+
+    var self = this;
+
+    var rowsPerCall = 100;
+    var processed = rowsPerCall;
+
+    async.whilst(
```

Comment:
I would ditch the usage of `async.whilst()` here, and just call yourself repeatedly.  e.g.

```javascript
Moray.prototype._reindexBucket = function _reindexBucket(bucketName, callback)
{
    assert.string(bucketName, 'bucketName');
    assert.func(callback, 'callback');

    var self = this;

    self.connection.reindexObjects(bucketName, 100, { noCache: true },
      function (err, res) {
        if (err || res.processed < 1) {
            callback(err);
            return;
        }

        self._reindexBucket(bucketName, callback);
    });
};
```